### PR TITLE
Added optional feature config to conditionally bypass CommitModal

### PR DESCRIPTION
**Pull request by calebjacob**
Related to: https://github.com/near/near-discovery/issues/787

- Added optional `commitModalBypass` feature config. When the `<CommitButton />` component is used inside of a widget with a matching `src` prop, the `CommitModal` will be bypassed and `onCommit()` will be called instantly when the button is clicked. If for some reason the requested transaction is invalid, the `CommitModal` will still appear to show an error message to the user. View example below to see configuration options.
- Added optional `enableWidgetSrcWithCodeOverride` boolean feature flag. This is helpful to enable when developing in a local environment when using a `redirectMap` in combination with the new `commitModalBypass` feature. With this enabled, any widget that is overridden via `redirectMap` can still reference its `src` prop to respect your `commitModalBypass` config.

```js
initNear({
  features: {
    commitModalBypass: {
      authorIds: ["mob.near", "root.near"], // Bypass for all widgets published by these accounts
      sources: [
        "cool.near/widget/CoolComponent",
        "awesome.near/widget/AwesomeComponent",
      ], // Bypass for specific components
    },
    enableWidgetSrcWithCodeOverride: isLocalEnvironment,
  },
});
```

### CODE REVIEW

**Review by charleslavon — APPROVED** [review]
LGTM. Tested bypassing entire accounts and a couple specific components. @evgenykuzyakov can you take a look?

File: src/lib/components/Widget.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ export const Widget = React.forwardRef((props, forwardedRef) => {
       setSrc(src);
     } else if (srcOrCode?.code) {
       setCode(srcOrCode.code);
-      setSrc(null);
+      setSrc(near.features.enableWidgetSrcWithCodeOverride ? propsSrc : null);
```
Comment (evgenykuzyakov) [inline src/lib/components/Widget.js:105]: This may throw undefined error

Comment (calebjacob) [reply src/lib/components/Widget.js:105]: I believe `near.features` will always be an object thanks to the default here in `near.js`:

Screen Shot 2024-01-03 at 11 01 50 AM

Comment (calebjacob) [reply src/lib/components/Widget.js:105]: I'll open up a PR to convert to `near?.features?.enableWidgetSrcWithCodeOverride` just to be extra safe.

Comment (calebjacob) [reply src/lib/components/Widget.js:105]: @evgenykuzyakov https://github.com/NearSocial/VM/pull/163

**Pull request merged**